node/node.go

```diff
@@ -350,6 +365,51 @@ func activateShhService(stack *node.Node, config *params.NodeConfig, db *leveldb
 	})
 }
 
+// activateWakuService configures Waku and adds it to the given node.
+func activateWakuService(stack *node.Node, config *params.NodeConfig, db *leveldb.DB) (err error) {
+	if !config.WakuConfig.Enabled {
+		logger.Info("Waku protocol is disabled")
+		return nil
+	}
+
+	err = stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
+		return createWakuService(ctx, &config.WakuConfig, &config.ClusterConfig)
+	})
+	if err != nil {
+		return
+	}
+
+	// Register eth-node node bridge
+	// TODO: what to do with it? Shouldn't it be configured when a node is created?
```

Comment:
cc @PombeirP

Reply:
Yeah, this should move out of `activateShhService`/`activateWakuService` because it's not really related to Whisper/Waku.